### PR TITLE
Remove state «almost_done» from withdraw and dispatch failed API calls to «failed» state (closes #476).

### LINKED ISSUE #476: Need to fix travis CI after pr #466 merge
https://github.com/rubykube/peatio/pull/466

### PR DESCRIPTION
**Pull request by ysv**

### CODE REVIEW

File: app/models/worker/withdraw_coin.rb [hunk 1]
```diff
@@ -42,6 +30,9 @@ def process(payload, metadata, delivery_info)
           withdraw.succeed
           withdraw.save!
         end
+
+        # TODO: (yssavchuk) implement #coin_api_withdrawal_error where log error and call withdraw.fail
```
Comment (ysv) [inline app/models/worker/withdraw_coin.rb]: @yivo your suggestions ?

File: lib/coin_api/bit_go.rb [hunk 2]
```diff
@@ -72,6 +72,8 @@ def rest_api(verb, path, data = nil)
 
       response = Faraday.send(verb, *args).assert_success!
       JSON.parse(response.body)
+    rescue Faraday::Error => e
```
Comment (ysv) [inline lib/coin_api/bit_go.rb]: @yivo do wee need to define CoinAPI::WithdrawalError or something like this processing this scenario  ?

Comment (yivo) [reply lib/coin_api/bit_go.rb]: @yssavchuk I am sure no. This makes no sense.

File: app/models/worker/withdraw_coin.rb (hunk 1 above)
Comment (yivo) [reply app/models/worker/withdraw_coin.rb]: @yssavchuk I think we can omit all rescues by wrapping whole process in single transaction. So we open transaction only if withdraw.processing?, we process it and if it fails DB will rollback us to `processing` so we will retry it. Just think simple.

File: app/models/worker/withdraw_coin.rb [hunk 3]
```diff
@@ -3,22 +3,10 @@ class WithdrawCoin
...
-      end
-
-      Withdraw.transaction do
-        withdraw = Withdraw.lock.find payload[:id]
-
-        return unless withdraw.almost_done?
+        return unless withdraw.processing
 
         balance = CoinAPI[withdraw.currency.to_sym].load_balance!
         raise Account::BalanceError, 'Insufficient coins' if balance < withdraw.sum
```
Comment (yivo) [inline app/models/worker/withdraw_coin.rb]: @yssavchuk Instead of raising error you can set status `suspect` for this case.

**Review by yivo — CHANGES_REQUESTED** [review]

**Review by yivo — APPROVED** [review]

**Pull request merged**